modules/eks/submodules/privatelink/vpc-endpoint-services.tf

```diff
@@ -0,0 +1,85 @@
+locals {
+  endpoint_services = { for service in var.vpc_endpoint_services : service.name => service.private_dns }
+
+  listeners = distinct(flatten([
+    for service in var.vpc_endpoint_services : [
+      for port in service.ports : {
+        service  = service.name
+        port     = port
+        cert_arn = service.cert_arn
+      }
+    ]
+  ]))
+}
+
+data "aws_route53_zone" "hosted" {
+  count        = var.route53_hosted_zone_name != null ? 1 : 0
+  name         = var.route53_hosted_zone_name
+  private_zone = false
+}
+
+resource "aws_lb" "nlbs" {
```

Comment:
I guess this is a broader question, but do we need a single NLB per vpce or can we have one shared NLB and vpce for all endpoints?

Reply:
currently amazon does not allow to select the same NLB for multiples VPC endpoint services